packages/ses/README.md

```diff
@@ -503,6 +502,194 @@ system generated other diagnostic information hidden in side tables. The tamed
+  as they would had they been given a reference to a working `Date.now()`.
+- be able to communicate, as they would if they had shared access to mutable
+  state like an unfrozen object, a hardened collection like a `Map`, or even
+  `Math.random()`.
+
+```js
+// Claims, Figure 2
+lockdown();
+const compartment = new Compartment();
+harden(compartment.globaThis);
+compartment.evaluate(program1);
+compartment.evaluate(program2);
+```
+
+However such programs (`program`, `program1`, nor `program2`) are only
+as useful as a calculator.
+A host program is therefore responsible for maintaining any of the desired
+invariants above when "endowing" a compartment with any of its own objects.
+
+For example, a host program (Figure 3) may run two programs in separate
+compartments, giving one program the ability to resolve a promise and the other
+program the ability to observe the settlement (fulfillment or rejection) of
+that promise.
+The host program is responsible for hardening these objects.
+
+```js
+// Claims, Figure 2
+lockdown();
+
+const promise = new Promise(resolve => {
+  const compartmentA = new Compartment(harden({
+    resolve,
+  }));
+  compartment.evaluate(programA);
+});
+
+const compartmentB = new Compartment(harden({
+  promise,
+}));
+compartmentB.evaluate(programB);
+```
+
+With `ses`, guest programs are initially powerless.
+A host can explicitly share limited powers with guest programs
+and provide intentional communication channels between them.
+
+Host programs must maintain the `ses` boundary with care in what they present
+as endowments.
+A host program should take care not to share mutable state with guests,
+or distribute mutable state to multiple guests, such as an unfrozen object (use
+`harden`), direct read and write access to a collection, like a `Map` or `Set`,
+even if hardened.
+Furthermore, typed arrays are collections and cannot be hardened.
+
+For the purposes of sharing state, pseudo-random number generators (PRNG) like
+`Math.random()` are equivalent to read and write access to shared state, and
+any guest can use one to eavesdrop on other guests or the host that share one.
+
+If a guest program needs a high resolution timer to function, the host should
+only invite one guest to a single operating system process and limit the
+activity of the host program in the same process.
+
+Hosts must avoid exposing `SharedArrayBuffer` to guests.
+Any two JavaScript programs sharing a `SharedArrayBuffer` can use the shared
+buffer to construct a high resolution timer.
+
+The `ses` shim does not in itself isolate the stack of guest programs, even
+when evaluated in separate compartment.
+This is relevant when program created objects are shared between guest
+programs.
+
+Any object endowed to a program (either through the per-compartment
+`globalThis`, function arguments or returned values) represents a potential
+risk, such as reentrancy, when it interacts with a guest object.
```

Comment:
```suggestion
When a program interacts with an object introduced by another program
(e.g. through the per-compartment `globalThis`, function arguments or
returned values), there are potential risks due to the synchronous nature
of object access.
```

Reply:
Adopted, but avoided Latin `e.g.,`.

Reply:
We lost the second sentence which I believe is still valuable.

Reply:
Oh, my mistake again. Pushed.